core/src/main/scala/quasar/destination/gbq/GBQDestination.scala

```diff
@@ -0,0 +1,244 @@
+        .withEntity(dCfg)
+
+      client.run(datasetReq).use { resp =>
+        resp.status match {
+          case Status.Ok =>
+            ().asRight[InitializationError[Json]].pure[F]
+          case Status.Conflict =>
+            DestinationError.invalidConfiguration(
+              (destinationType,
+              jString(s"Reason: gbq dataset ${resp.status.reason}"),
+              ZNEList(resp.status.reason))).asLeft.pure[F]
+          case status =>
+            DestinationError.malformedConfiguration(
+              (destinationType, jString("Reason: " + status.reason), 
+              sanitizedConfig.toString)).asLeft.pure[F]
+        }
+      }
+    }
+
+  private def mkGbqJob(client: Client[F], accessToken: String, jCfg: GBQJobConfig): F[Either[InitializationError[Json], Uri]] = {
+    implicit def jobConfigEntityEncoder: EntityEncoder[F, GBQJobConfig] = jsonEncoderOf[F, GBQJobConfig]
+
+    val authToken = Authorization(Credentials.Token(AuthScheme.Bearer, accessToken))
+    val jobReq = Request[F](
+      method = Method.POST,
+      uri = Uri
+        .fromString(s"https://bigquery.googleapis.com/upload/bigquery/v2/projects/${jCfg.destinationTable.project}/jobs?uploadType=resumable")
+        .getOrElse(Uri()))
+        .withHeaders(authToken)
+        .withContentType(`Content-Type`(MediaType.application.json))
+        .withEntity(jCfg)
+
+    client.run(jobReq).use { resp =>
+      resp.status match {
+        case Status.Ok => resp.headers match {
+          case Location(loc) =>
+            loc.uri.asRight.pure[F]
+        }
+        case _ =>
+          DestinationError.malformedConfiguration(
+            (destinationType, jString("Reason: " + resp.status.reason), 
+            sanitizedConfig.toString)).asLeft.pure[F]
+      }
+    }
+  }
+
+  private def upload(client: Client[F], bytes: Stream[F, Byte], uploadLocation: Uri): Stream[F, Unit] = {
+    val destReq = Request[F](method = Method.PUT, uri = uploadLocation)
+        .putHeaders(Header("Host", "www.googleapis.com"))
+        .withContentType(`Content-Type`(MediaType.application.`x-www-form-urlencoded`))
+        .withEntity(bytes)
+    val doUpload = client.run(destReq).use { resp =>
+      resp.status match {
+        case Status.Ok => ().pure[F]
+        case _ => ApplicativeError[F, Throwable].raiseError[Unit](
+            new RuntimeException(s"Upload failed: ${resp.status.reason}" ))
+      }
+    }
+    Stream.eval(doUpload)
+  }
+}
+
+object GBQDestination {
+  def apply[F[_]: Concurrent: ContextShift: MonadResourceErr: ConcurrentEffect, C](client: Client[F], config: GBQConfig, sanitizedConfig: Json)
+      : Resource[F, Either[InitializationError[C], LegacyDestination[F]]] = {
+        val gbqDest: Either[InitializationError[C], LegacyDestination[F]] = 
+          new GBQDestination[F](client, config, sanitizedConfig).asRight[InitializationError[C]]
+          
+        Resource.liftF(gbqDest.pure[F])
```

Comment:
porting over @djspiewak 's question from before:
> Why is this a Resource? Is there anything in GBQDestination which needs to be cleaned up which isn't being cleaned up?

Reply:
`GBQDestination` does make a few http requests, how can I tell if those aren't being cleaned up.

Reply:
Here's the `Resource.make` constructor. There are a few different ways to construct a `Resource`; this signature captures the intent of the `Resource` data structure.
```scala
def make[F[_], A](acquire: F[A])(release: A => F[Unit])(implicit F: Functor[F]): Resource[F, A]
```


The `acquire: F[A]` is an effectful program you're running, which ultimately outputs a value of type `A`. The `release: A => F[Unit]` returns a semantically valueless effect (`F[Unit]`) that will run after the `F[A]` completes. In practice, the `F[A]` will be used to _acquire_ something without actually performing useful computation. This is where `Resource#use` comes in:
```scala
def use[G[x] >: F[x], B](f: A => G[B])(implicit F: Bracket[G[?], Throwable]): G[B]
```
Given a `Resource`, you can `use` it. The _how_ of how you want to use it is embodied in the `f: A => G[B]` function. 

Note that `use` introduces a new type `B`, but you can only define how to release the resource (`release: A => F[Unit]`) given a value of type `A`. This separates the concerns of acquiring/releasing the resource from the usage of the resource. 

Let's consider another way to create a `Resource`: `Resource.liftF`.
```scala
def liftF[F[_], A](fa: F[A])(implicit F: Applicative[F]): Resource[F, A]
```
This signature is missing the critical component of what makes `Resource` what it is namely, a way to `release` this resource. So why is this even useful? If you're in a context where you already have a `Resource`, you might need to call `Resource.liftF` to "lift" another value/effect into this context so you can `flatMap` over it.

For that reason, seeing a `Resource.liftF` by itself here is a cause for concern, because it isn't being used to release (i.e. clean up) anything. So that means either 1) we should be using `Resource.make` to provide a `release` or 2) we can just return `F[Either[InitializationError[C], LegacyDestination[F]]]` and not use `Resource` at all.

Here are a couple examples of `Resource.make` being used in the postgres destination.
- https://github.com/precog/quasar-destination-postgres/blob/c9243ddf77277f9cf31025c0bebe063fa08c7a4c/core/src/main/scala/quasar/plugin/postgres/PostgresDestinationModule.scala#L118
- https://github.com/precog/quasar-destination-postgres/blob/c9243ddf77277f9cf31025c0bebe063fa08c7a4c/core/src/main/scala/quasar/plugin/postgres/PostgresDestinationModule.scala#L151

Reply:
Also, http requests via `http4s` will not leave open any resources that need to be cleaned up, as the resource management is handled in the `Stream`. So, as far as I've seen, we can just return `F[Either[InitializationError[C], LegacyDestination[F]]]` here.

Reply:
The signature of `DestinationModule#destination` returns `Resource[F, Either[InitializationError[Json], Destination[F]]]`. So we will have to lift into `Resource` somewhere (and that somewhere is actually likely here). 

This signature gives a destination a `Resource` to work within, if it needs to acquire+release resources. But not all destinations need to do that.